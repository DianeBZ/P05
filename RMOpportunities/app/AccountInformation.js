import React, { Component } from 'react';
import Translation from './Translation';

import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  ListView
} from 'react-native';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

//We change this and add some foncs to realise
//the display of the list of factures when we need.
var dataSource = null;

export default class AccountInformation extends Component {
  constructor(props) {
  super(props);
  var ds1 = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
  var ds2 = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
  var ds3 = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
  
  this.state = {
	dataSource1: ds1.cloneWithRows([Translation[lang].denomination, Translation[lang].adresse , Translation[lang].num_tva_inter]),
    dataSource2: ds1.cloneWithRows(['NULL1', 'NULL2',"NULL3"]),
	dataSource3: ds1.cloneWithRows([Translation[lang].nom_info , Translation[lang].ad_mail , Translation[lang].num_tel]),
  };

}
	  
  render() {
    return (
      <View style={styles.container}>
        <View style={{height:height*0.08}}/>
    		<View style={styles.containerMain}>

    			<View style = {styles.containerText}>
    				<Text style = {styles.title}>
    					{Translation[lang].info}
    				</Text>
    				<Text style = {styles.instructions}>
    					{Translation[lang].text_info}
    				</Text>
    			</View>


          <View style = {{justifyContent: 'flex-start',alignSelf: 'flex-start',marginLeft:5}}>
            <Text style = {styles.titleAV}>
                {Translation[lang].entreprise}
            </Text>
          </View>
          <View style={styles.containerEntreprise}>
            <View style={{flex:1}}>
            <ListView
              contentContainerStyle={styles.containerEntrepriseGauche}
              dataSource={this.state.dataSource1}
              renderRow={(rowData) => <Text>{rowData}</Text>}/>
            </View>
            <View style={{flex:1.8}}>
              <ListView
              contentContainerStyle={styles.containerEntrepriseDroite}
              dataSource={this.state.dataSource2}
              renderRow={(rowData) => <Text>{rowData}</Text>}/>
            </View>
          </View>

          <View style = {{justifyContent: 'flex-start',alignSelf: 'flex-start',marginLeft:5}}>
            <Text style = {styles.titleAV}>
                {Translation[lang].representant}
            </Text>
          </View>
          <View style={styles.containerEntreprise}>
            <View style={{flex:1}}>
            <ListView
              contentContainerStyle={styles.containerEntrepriseGauche}
              dataSource={this.state.dataSource3}
              renderRow={(rowData) => <Text>{rowData}</Text>}/>
            </View>
            <View style={{flex:1.8}}>
              <ListView
              contentContainerStyle={styles.containerEntrepriseDroite}
              dataSource={this.state.dataSource2}
              renderRow={(rowData) => <Text>{rowData}</Text>}/>
            </View>
          </View>



    		</View>
    </View>


    );
  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  containerMain: {
    backgroundColor: '#F5FCFF',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.89,
	alignSelf: 'center',
  },
  containerText: {
    backgroundColor: '#F5FCFF',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.28,
	alignSelf: 'center',
  },
  title: {
    fontSize: 40,
    textAlign: 'center',
    margin: 10,
	color: '#333333',
  },
  instructions: {
    fontSize: 15,
	textAlign: 'center',
    marginBottom:5,
  },
  titleAV: {
  fontSize: 20,
  color: '#333333',
  textAlign: 'left',
  },
  containerEntreprise: {
  //width: width,
  flexDirection:'row',
  height: height * 0.23,
  //justifyContent: 'space-around',
  //alignItems: 'stretch',
  backgroundColor: '#F5FCFF',
  padding: 5,
  borderWidth: 1,
  borderRadius: 5,
  margin: 5,
  borderColor: '#CCC',
  width: 0.92 * width + 20,
  },
  containerEntrepriseGauche:{
      flex:1,
      justifyContent:'space-around'
    //alignItems:'flex-end',
  },
  containerEntrepriseDroite:{
      flex:1,
      justifyContent:'space-around',
      alignItems:'flex-end'
  }
});