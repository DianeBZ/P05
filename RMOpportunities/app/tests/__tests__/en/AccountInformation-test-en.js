// __tests__/AccountInformation-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import AccountInformation from '../../../main/AccountInformation';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <AccountInformation />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
