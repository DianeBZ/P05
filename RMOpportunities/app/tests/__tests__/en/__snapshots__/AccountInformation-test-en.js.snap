// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "height": 106.72,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#F2F2F2",
        "flexDirection": "column",
        "flexGrow": 0,
        "flexShrink": 0,
        "height": 1187.26,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F2F2F2",
          "flexDirection": "column",
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 373.52000000000004,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 40,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Information
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "marginBottom": 5,
            "textAlign": "center",
          }
        }
      >
        View your account related information
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "justifyContent": "flex-start",
          "marginLeft": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 20,
            "textAlign": "left",
          }
        }
      >
        Company
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "borderColor": "#CCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 306.82,
          "margin": 5,
          "padding": 5,
          "width": 710,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "flex": 1,
              "justifyContent": "space-around",
            }
          }
          dataSource={
            ListViewDataSource {
              "items": 3,
            }
          }
          renderRow={[Function]}
          renderScrollComponent={[Function]}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Name
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Address
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              VAT identification number
            </Text>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "flex": 1.8,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "justifyContent": "space-around",
            }
          }
          dataSource={
            ListViewDataSource {
              "items": 3,
            }
          }
          renderRow={[Function]}
          renderScrollComponent={[Function]}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              NULL1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              NULL2
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              NULL3
            </Text>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "justifyContent": "flex-start",
          "marginLeft": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 20,
            "textAlign": "left",
          }
        }
      >
        Representative
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "borderColor": "#CCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 306.82,
          "margin": 5,
          "padding": 5,
          "width": 710,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "flex": 1,
              "justifyContent": "space-around",
            }
          }
          dataSource={
            ListViewDataSource {
              "items": 3,
            }
          }
          renderRow={[Function]}
          renderScrollComponent={[Function]}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Name
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Email address
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Phone number
            </Text>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "flex": 1.8,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "justifyContent": "space-around",
            }
          }
          dataSource={
            ListViewDataSource {
              "items": 3,
            }
          }
          renderRow={[Function]}
          renderScrollComponent={[Function]}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              NULL1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              NULL2
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              NULL3
            </Text>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
